lib/jpegli/decode_api_test.cc

```diff
@@ -109,434 +64,229 @@ class ChunkedSourceManager : public SourceManager {
     src->pos_ += src->pub_.bytes_in_buffer;
     return TRUE;
   }
+
+  static void skip_input_data(j_decompress_ptr cinfo, long num_bytes) {}
+
+  static void term_source(j_decompress_ptr cinfo) {}
 };
 
-class SuspendingSourceManager : public SourceManager {
- public:
-  SuspendingSourceManager(const uint8_t* data, size_t len,
-                          size_t max_chunk_size)
-      : SourceManager(data, len, max_chunk_size) {
-    pub_.init_source = init_source;
-    pub_.fill_input_buffer = fill_input_buffer;
+void SetDecompressParams(const DecompressParams& dparams,
+                         j_decompress_ptr cinfo) {
+  jpegli_set_output_format(cinfo, dparams.data_type, dparams.endianness);
+  cinfo->raw_data_out = dparams.output_mode == RAW_DATA;
+  cinfo->do_block_smoothing = dparams.do_block_smoothing;
+  if (dparams.set_out_color_space) {
+    cinfo->out_color_space = dparams.out_color_space;
+    if (dparams.out_color_space == JCS_UNKNOWN) {
+      cinfo->jpeg_color_space = JCS_UNKNOWN;
+    }
   }
+}
 
-  bool LoadNextChunk() {
-    if (pos_ >= len_) {
-      return false;
+void ReadOutputImage(const DecompressParams& dparams, j_decompress_ptr cinfo,
```

Comment:
Perhaps `ReadCroppedOutputImage`?
There is another `ReadOutputImage` that looks much shorter...

Reply:
This is sort-of intentional, all of the decoding tests have a ReadOutputImage that uses the decompress parameters relevant for that specific test. It is there to deduplicate this code for the different test cases within the test.cc file.